packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -118,3 +124,50 @@ async function assertNoBlocks(
     )
   }
 }
+
+const ensureReadAfterWrite = async (
+  ctx: AppContext,
+  requester: string,
+  res: ApiRes<OutputSchema>,
+): Promise<OutputSchema> => {
+  const clock = getClock(res.headers)
+  if (!clock) return res.data
+  const author = getAuthor(res)
+  if (author !== requester) return res.data
+  const localSrvc = ctx.services.local(ctx.db)
+  const local = await localSrvc.getRecordsSinceClock(requester, clock, [
+    ids.AppBskyActorProfile,
+  ])
+  const localProf = local.profile
+  if (!localProf) return res.data
+  const feed = res.data.feed.map((item) => {
+    if (item.post.author.did === requester) {
+      return {
+        ...item,
+        post: {
+          ...item.post,
+          author: localSrvc.updateProfileViewBasic(
+            item.post.author,
+            localProf.record,
+          ),
+        },
+      }
+    } else {
+      return item
+    }
+  })
+  return {
+    ...res.data,
+    feed,
+  }
+}
+
+const getAuthor = (res: ApiRes<OutputSchema>): string | undefined => {
```

Comment:
I take it this is a little workaround so that we don't have to potentially resolve a handle, does that seem right?

Reply:
yea basically 👍

imo we basically want to work off of the response + local records as much as possible